crates/sovereign-sdk/module-system/sov-modules-stf-blueprint/src/stf_blueprint.rs

```diff
@@ -129,48 +128,39 @@ where
                     };
 
                     tx_receipts.push(receipt);
-                    continue;
+
+                    return Err(StateTransitionError::HookError(e));
                 }
             };
             // Commit changes after pre_dispatch_tx_hook
-            sc_workspace = sc_workspace.checkpoint().to_revertable();
+            // sc_workspace = sc_workspace.checkpoint().to_revertable();
```

Comment:
why is this not reverted in case of error now?

Reply:
we upstream error and expect whoever running this soft confirmation to discard any changes made by the soft confirmation until the error point